catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -0,0 +1,303 @@
+        this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+        this.resolver = (resolver == null) ? new DynamicSchemaResolver() : resolver;
+
+    }
+
+    public RemoteSolrCatalogProvider(FilterAdapter filterAdapter, SolrClient client,
+            SolrFilterDelegateFactory solrFilterDelegateFactory) {
+        this(filterAdapter, client, solrFilterDelegateFactory, null);
+    }
+
+    public RemoteSolrCatalogProvider(FilterAdapter filterAdapter,
+            SolrFilterDelegateFactory solrFilterDelegateFactory) {
+        this(filterAdapter, null, solrFilterDelegateFactory, null);
+        updateClient();
+    }
+
+    @Override
+    public void maskId(String id) {
+        super.maskId(id);
+        provider.maskId(id);
+    }
+
+    /**
+     * Used to signal to the Solr client to commit on every transaction. Updates
+     * the underlying ConfigurationStore so that the property is propagated
+     * throughout the Solr Catalog Provider code
+     *
+     * @param forceAutoCommit
+     */
+    public void setForceAutoCommit(boolean forceAutoCommit) {
+        ConfigurationStore.getInstance()
+                .setForceAutoCommit(forceAutoCommit);
+    }
+
+    public void setDisableTextPath(boolean disableTextPath) {
+        ConfigurationStore.getInstance()
+                .setDisableTextPath(disableTextPath);
+    }
+
+    @Override
+    public Set<ContentType> getContentTypes() {
+        return getProvider().getContentTypes();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return getProvider().isAvailable();
+    }
+
+    @Override
+    public boolean isAvailable(SourceMonitor callback) {
+        return getProvider().isAvailable(callback);
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest queryRequest) throws UnsupportedQueryException {
+        return getProvider().query(queryRequest);
+    }
+
+    @Override
+    public String getDescription() {
+        return describableProperties.getProperty("description", "");
+    }
+
+    @Override
+    public String getOrganization() {
+        return describableProperties.getProperty("organization", "");
+    }
+
+    @Override
+    public String getTitle() {
+        return describableProperties.getProperty("name", "");
+    }
+
+    @Override
+    public String getVersion() {
+        return describableProperties.getProperty("version", "");
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest createRequest) throws IngestException {
+        return getProvider().create(createRequest);
+    }
+
+    @Override
+    public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException {
+        return getProvider().delete(deleteRequest);
+    }
+
+    @Override
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException {
+        return getProvider().update(updateRequest);
+    }
+
+    public void shutdown() {
+        closeSolrClient();
+    }
+
+    /**
```

Comment:
Should we move this comment to the `shutdown` method since it's `public` and this one isn't?